build-homebrew-package.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+set -x
+
+if [ -z "${HZ_DISTRIBUTION}" ]; then
+  echo "Variable HZ_DISTRIBUTION is not set. It must be set to 'hazelcast' for OS, 'hazelcast-enterprise' for EE"
+  exit 1
+fi
+
+if [ -z "${HZ_VERSION}" ]; then
+  echo "Variable HZ_VERSION is not set. This is the version of Hazelcast used to build the package."
+  exit 1
+fi
+
+if [ -z "${PACKAGE_VERSION}" ]; then
+  echo "Variable PACKAGE_VERSION is not set. This is the version of the built package."
+  exit 1
+fi
+export PACKAGE_VERSION=$(echo $PACKAGE_VERSION | tr '[:upper:]' '[:lower:]' | sed -r -r 's/(-)/\./g')
+
+echo "Building Homebrew package $HZ_DISTRIBUTION:${HZ_VERSION} package version ${PACKAGE_VERSION}"
+
+
+if [[ "${HZ_VERSION}" == *"SNAPSHOT"* ]]; then
+  if [ "${HZ_DISTRIBUTION}" == "hazelcast" ]; then
+    export MAVEN_REPO="https://oss.sonatype.org/content/repositories/snapshots"
+  else
+    export MAVEN_REPO="https://repository.hazelcast.com/snapshot"
+  fi
+
+  curl -O -fsSL "$MAVEN_REPO/com/hazelcast/${HZ_DISTRIBUTION}-distribution/${HZ_VERSION}/maven-metadata.xml"
+  TIMESTAMP=$(xmllint --xpath "/metadata/versioning/snapshotVersions/snapshotVersion[1]/value/text()" maven-metadata.xml)
+
+  export PACKAGE_URL="$MAVEN_REPO/com/hazelcast/${HZ_DISTRIBUTION}-distribution/${HZ_VERSION}/${HZ_DISTRIBUTION}-distribution-${TIMESTAMP}.tar.gz"
+elif [[ "${HZ_VERSION}" == *"DR"* ]]; then
+  export MAVEN_REPO="https://repository.hazelcast.com/devel"
+  export PACKAGE_URL="$MAVEN_REPO/com/hazelcast/${HZ_DISTRIBUTION}-distribution/${HZ_VERSION}/${HZ_DISTRIBUTION}-distribution-${HZ_VERSION}.tar.gz"
+else
+  if [ "${HZ_DISTRIBUTION}" == "hazelcast" ]; then
+    export MAVEN_REPO="https://repo1.maven.org/maven2"
+  else
+    export MAVEN_REPO="https://repository.hazelcast.com/release"
+  fi
+  export PACKAGE_URL="$MAVEN_REPO/com/hazelcast/${HZ_DISTRIBUTION}-distribution/${HZ_VERSION}/${HZ_DISTRIBUTION}-distribution-${HZ_VERSION}.tar.gz"
+fi
```

Comment:
This is quite complicated. I wonder if we could ask maven for the package URL. A bit hacky way would be something like:
```bash
mvn dependency:get -Dartifact=-Dartifact=com.hazelcast:${HZ_DISTRIBUTION}-distribution:${HZ_VERSION}:tar.gz -Dtransitive=false | grep 'Downloaded from' | grep -Eo 'https://[^ >]+'
```
What do you think?